java/server/src/org/openqa/selenium/grid/distributor/Distributor.java

```diff
@@ -250,17 +268,19 @@ public CreateSessionResponse newSession(HttpRequest request)
         EventAttribute.setValue("Unable to create session: " + e.getMessage()));
       span.addEvent(AttributeKey.EXCEPTION_EVENT.getKey(), attributeMap);
 
-      throw e;
+      return Either.left(e);
     } catch (IOException e) {
       span.setAttribute("error", true);
       span.setStatus(Status.UNKNOWN);
 
       EXCEPTION.accept(attributeMap, e);
-      attributeMap.put(AttributeKey.EXCEPTION_MESSAGE.getKey(),
-        EventAttribute.setValue("Unknown error in LocalDistributor while creating session: " + e.getMessage()));
+      attributeMap.put(
```

Comment:
Maybe reverting this change will make the review easier. It is changing the formatting.

Reply:
+1